src/App.tsx

```diff
@@ -1,93 +1,130 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useMemo, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { useLocalStorage } from './utils/useLocalStorage';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useLocalStorage('todos', []);
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const [completedTodo, uncompletedTodo] = todos.reduce(
+    (acc: Todo[][], todo: Todo) => {
+      if (todo.completed) {
+        acc[0].push(todo);
+      } else {
+        acc[1].push(todo);
+      }
+
+      return acc;
+    },
+    [[], []],
+  );
+
+  const handleAddTodo = (todo: Todo) => {
+    setTodos([...todos, todo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle.trim().length) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      title: todoTitle,
+      id: +new Date(),
+      completed: false,
+    };
+
+    handleAddTodo(newTodo);
+    setTodoTitle('');
+  };
+
+  const handleToggleCompleted = (todoId?: number) => {
+    setTodos(todos.map((todo: Todo) => (todo.id === todoId
+      ? { ...todo, completed: !todo.completed }
+      : todo)));
```

Comment:
its better to use prevSate
Fix all similar cases
```suggestion
    setTodos(prevTodos => prevTodos.map(...));
```